kuser/kinit.c

```diff
@@ -217,19 +217,61 @@ usage(int ret)
     exit(ret);
 }
 
+static krb5_error_code
+get_starting_tgs_name(krb5_context context,
+		      krb5_ccache ccache,
+		      krb5_principal *tgs_name)
+{
+    krb5_error_code ret;
+    krb5_data data;
+    char *realm;
+
+    ret = krb5_cc_get_config(context, ccache, NULL, "start_realm", &data);
+    if (ret)
+	return ret;
+
+    realm = malloc(data.length + 1);
+    memcpy(realm, data.data, data.length);
+    realm[data.length] = '\0';
+
+    ret = krb5_make_principal(context, tgs_name, realm,
+			      KRB5_TGS_NAME, realm, NULL);
+    free(realm);
+
```

Comment:
Don't we need to free the krb5_data allocated by krb5_cc_get_config() -> krb5_copy_data()?

`free(data);`